### PR TITLE
Tidy update

### PR DESCRIPTION
**Pull request by shurale-nkn**
Static checks update for modern rocm hip-clang.

Stages with HCC usage and hcc dockerfile was removed.
Docker image building now does not prevent the task from termination.

### CODE REVIEW

File: src/conv/invokers/impl_gemm_dynamic.cpp [hunk 1]
```diff
@@ -94,7 +94,7 @@ float CallImplGemmDynamicForward1x1(const miopen::Handle& handle,
     int dilation_w  = conv_problem.GetDilationW();
     int pad_h       = conv_problem.GetPadH();
     int pad_w       = conv_problem.GetPadW();
-    int __pack0     = 0;
+    int __pack0     = 0; //NOLINT (bugprone-reserved-identifier)
```
Comment (pfultz2) [inline src/conv/invokers/impl_gemm_dynamic.cpp:97]: This variable should be renamed.

File: src/include/miopen/mlo_utils.hpp [hunk 2]
```diff
@@ -84,7 +84,7 @@ typedef unsigned int uint;
 #include <mach/mach.h>
 #include <mach/mach_time.h>
 #endif
-using __int64 = long long;
+using __int64 = long long; // NOLINT (bugprone-reserved-identifier)
```
Comment (pfultz2) [inline src/include/miopen/mlo_utils.hpp:87]: This type should be renamed as it can conflict with identifiers in the runtime.

File: test/reduce_test.cpp [hunk 3]
```diff
@@ -777,8 +779,8 @@ struct reduce_driver : test_driver
             assert(toReduceDims[i] < inLengths.size());
 
         // set the lengths of the dimensions to be reduced to 1 to represent the output Tensor
-        for(int i                       = 0; i < toReduceDims.size(); i++)
-            outLengths[toReduceDims[i]] = static_cast<std::size_t>(1);
+        for(int& toReduceDim : toReduceDims)
```
Comment (pfultz2) [inline test/reduce_test.cpp:782]: This should be `const int&` or just `int`.

File: CMakeLists.txt [hunk 4]
```diff
@@ -431,22 +431,26 @@ include(ClangTidy)
...
+        -cert-dcl37-c
+        -cert-dcl51-cpp
         -clang-analyzer-alpha.core.CastToStruct
         -clang-analyzer-optin.performance.Padding
         -clang-diagnostic-deprecated-declarations
         -clang-diagnostic-extern-c-compat
         -clang-diagnostic-unused-command-line-argument
         -cppcoreguidelines-avoid-c-arrays
         -cppcoreguidelines-avoid-magic-numbers
+        -cppcoreguidelines-avoid-non-const-global-variables
```
Comment (pfultz2) [inline CMakeLists.txt:453]: This seems like an important check that we should not disable. Why is this necessary?

Comment (shurale-nkn) [reply CMakeLists.txt:453]: Because I thought we were using a lot of global variables like " extern bool"
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f8f7e325cd2a9c2e0a7e3944f226c8b800b87968/src/include/miopen/logger.hpp#L202
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f8f7e325cd2a9c2e0a7e3944f226c8b800b87968/src/include/miopen/solver.hpp#L53
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f8f7e325cd2a9c2e0a7e3944f226c8b800b87968/src/include/miopen/find_controls.hpp#L41
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f8f7e325cd2a9c2e0a7e3944f226c8b800b87968/src/include/miopen/find_db.hpp#L64

https://github.com/ROCmSoftwarePlatform/MIOpen/blob/f8f7e325cd2a9c2e0a7e3944f226c8b800b87968/src/solver/conv_direct_naive_conv.cpp#L38

but there were only 5 of them

Comment (shurale-nkn) [reply CMakeLists.txt:453]: I'll just add them to the exceptions

Comment (pfultz2) [reply CMakeLists.txt:453]: Can we and `NOLINT` to those lines instead?

Also, `AlwaysEnableConvDirectNaive` can be removed as its not used.

Comment (shurale-nkn) [reply CMakeLists.txt:453]: Yes, by exception I meant NOLINT

**Comment by atamazov** [comment]
Converted to draft to avoid spoiling docker image cache on CI. Please change this back to "normal" PR only after ending of CI moratorium (see mail). Then first merge from develop. Push the `[Ready for review]` button only after that. Thanks.

File: test/perfdb.cpp [hunk 5]
```diff
@@ -51,21 +51,25 @@ namespace tests {
...
+    // NOLINTNEXTLINE (cppcoreguidelines-avoid-non-const-global-variables)
     static std::mutex mutex;
     std::lock_guard<std::mutex> lock(mutex);
 
+    // NOLINTNEXTLINE (cppcoreguidelines-avoid-non-const-global-variables)
     static boost::optional<std::string> path(boost::none);
     return path;
 }
 
 static bool& full_set()
```
Comment (shurale-nkn) [inline test/perfdb.cpp:70]: Does anyone know if this feature is necessary? Or can it be removed?

**Comment by shurale-nkn** [comment]
static checks passed

File: test/perfdb.cpp (hunk 5 above)
Comment (atamazov) [reply test/perfdb.cpp:70]: @shurale-nkn please ask @DrizztDoUrden

**Comment by atamazov** [comment]
@shurale-nkn could you please merge latest develop? It contains massive change (Fwd GEMM invokers) and may hide some problems that will appear with this tidy update.

**Comment by atamazov** [comment]
@shurale-nkn BTW we will need to ask everyone to update their PRs as soon as this merged.

File: src/solver/conv_asm_implicit_gemm_wrw_gtc_dynamic_xdlops.cpp [hunk 6]
```diff
@@ -44,6 +44,7 @@ GetImplicitGemmWrwGTCDynamicXdlopsKernelList()
 {
     // retrieve dynamic igemm wrw pass's possible kernel name
     // clang-format off
+    // NOLINTNEXTLINE (cppcoreguidelines-avoid-non-const-global-variables)
```
Comment (atamazov) [inline src/solver/conv_asm_implicit_gemm_wrw_gtc_dynamic_xdlops.cpp:47]: And here (and wherever similar cases appear)

File: test/pooling_common.hpp [hunk 7]
```diff
@@ -48,11 +48,15 @@
 #include "cpu_conv.hpp"
 
 #define TEST_PADDING_MODE 0
-
-static int num_uint16_case        = 0;
-static int num_uint32_case        = 0;
+// NOLINTNEXTLINE (cppcoreguidelines-avoid-non-const-global-variables)
```
Comment (atamazov) [inline test/pooling_common.hpp:51]: [Noice] Tests are cans of worms

File: driver/conv_driver.hpp [hunk 8]
```diff
@@ -125,19 +125,6 @@ struct AutoMiopenWarmupMode
     bool debug_find_enforce_disable_prev;
 };
 
-struct AutoConvDirectNaiveAlwaysEnable
```
Comment (atamazov) [inline driver/conv_driver.hpp:128]: Can you bring this back?

File: CMakeLists.txt (hunk 4 above)
Comment (atamazov) [reply CMakeLists.txt]: > Also, AlwaysEnableConvDirectNaive can be removed as its not used.

It is used.

File: test/perfdb.cpp (hunk 5 above)
Comment (atamazov) [reply test/perfdb.cpp:70]: @DrizztDoUrden ping

**Review by atamazov — CHANGES_REQUESTED** [review]
One blocking comment

File: test/perfdb.cpp (hunk 5 above)
Comment (DrizztDoUrden) [reply test/perfdb.cpp:70]: I am not sure if anyone commonly uses this now, but in ancient times I have used it to debug PlainTextDb in a more stressful test. Is there any specific reason to remove it? It helps to triage db MP/MT failures, but one can just change values manually for the same result.

Comment (atamazov) [reply test/perfdb.cpp:70]: @DrizztDoUrden @shurale-nkn It is used here, not Ok to remove:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/eae8583eb49db4dcc874d2eadee9051e70f9eae4/test/perfdb.cpp#L1290

Comment (shurale-nkn) [reply test/perfdb.cpp:70]: if true than true?)

Comment (atamazov) [reply test/perfdb.cpp:70]: Read carefully: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/eae8583eb49db4dcc874d2eadee9051e70f9eae4/test/perfdb.cpp#L1288-L1290

**Review by atamazov — APPROVED** [review]

File: test/perfdb.cpp (hunk 5 above)
Comment (shurale-nkn) [reply test/perfdb.cpp:70]: I'm not sure is it a good decision. We are using static global tests::full_set() only in DbMultiProcessReadTest().Run(); and DbMultiProcessTest().Run(); which are called by the 6 lines below in same function, which means we can pass it as argument.

Comment (shurale-nkn) [reply test/perfdb.cpp:70]: The problem is that these ... ambiguous solutions, become the rule, not the exception, and they begin to be thoughtlessly copied.

Comment (atamazov) [reply test/perfdb.cpp:70]: This decision is not good, but addressing it is out of the scope of this PR. /cc @DrizztDoUrden

**Review by DrizztDoUrden — APPROVED** [review]
LGTM

File: test/perfdb.cpp (hunk 5 above)
Comment (DrizztDoUrden) [reply test/perfdb.cpp:70]: If you are talking about static variable specifically rather than --all mode for this test in general, I am 100% ok with changing it to a method parameter.
